river/anomaly/mstream/record_hash.py

```diff
@@ -1,11 +1,79 @@
+import math
+import random
+from itertools import compress
+import numpy as np
+
 class RecordHash():
-    def __init__(self):
-        pass
+    def __init__(self, is_categorical, number_buckets):
+        self.is_categorical = is_categorical
+        self.is_numerical = [ not feature for feature in is_categorical ]
+        # self.num_rows = num_rows number of hash functions, we will just use 2 for now
+        self.number_buckets = number_buckets
+        self.categorical_hash = RecordCategorialHash()
+        self.numerical_hash = RecordNumericalHash()
+        self.dimension_1 = len(self.is_numerical) # number of numerical features
+        self.dimension_2 = len(self.is_categorical) # number of categorical features
 
     def insert(self, x):
-        pass
+        # divide x into categorical and numerical
+        x_categorical = compress(x, self.is_categorical)
+        x_numerical = compress(x, self.is_numerical)
+        self.categorical_hash.hash(x_categorical)
+        self.numerical_hash.hash(x_numerical)    
 
+    def get_count(self):
+        return (self.categorical_hash.get_count(), self.numerical_hash.get_count()) % self.number_buckets
 
 class RecordCategorialHash():
-    def __init__(self):
-        pass
+    def __init__(self, dimension_2, num_buckets):
+        self.cat_recordhash = []
+        self.dimension_2 = dimension_2
+        self.num_buckets = num_buckets
+        for i in range(0, self.dimension_2):
```

Comment:
```suggestion
        for i in range(self.dimension_2):
```
Should be the same, right?